adafruit_feather_nrf52840: tests/subsys/portability/cmsis_rtos_v2/portability.cmsis_rtos_v2 fails when building

**Describe the bug**

The test in `tests/subsys/portability/cmsis_rtos_v2/portability.cmsis_rtos_v2` fails to build for the following two Zephyr targets:
* `adafruit_feather_nrf52840/nrf52840/sense`
* `adafruit_feather_nrf52840/nrf52840/sense/uf2`

The following compilation error is reported:
```
/home/filip/zephyrproject/zephyr/drivers/sensor/voltage_divider/voltage.c:118:41: warning: implicit
declaration of function 'K_TIMEOUT_ABS_TICKS'; did you mean 'Z_TIMEOUT_MS_TICKS'? [-Wimplicit-function-declaration]
  118 |                 data->earliest_sample = K_TIMEOUT_ABS_TICKS(
      |                                         ^~~~~~~~~~~~~~~~~~~
      |                                         Z_TIMEOUT_MS_TICKS
/home/filip/zephyrproject/zephyr/drivers/sensor/voltage_divider/voltage.c:118:41: error: incompatible types when assigning to type 'k_timeout_t' from type 'int'
/home/filip/zephyrproject/zephyr/drivers/sensor/voltage_divider/voltage.c: In function 'voltage_ini
':
/home/filip/zephyrproject/zephyr/drivers/sensor/voltage_divider/voltage.c:145:33: error: incompatible types when assigning to type 'k_timeout_t' from type 'int'
  145 |         data->earliest_sample = K_TIMEOUT_ABS_TICKS(0);
```

**To Reproduce**
Steps to reproduce the behavior:
1. `west build -p -b adafruit_feather_nrf52840/nrf52840/sense tests/subsys/portability/cmsis_rtos_v2`
2. See error

**Expected behavior**

Build doesn't end with an error.

**Impact**

Showstopper (failing CI). See #80038.

**Environment (please complete the following information):**

 - OS: Linux
 - Toolchain: Zephyr SDK
 - Commit SHA or Version used: 185432c96dd74fb30edc334f973a5a42876d22f0

--- a/samples/subsys/portability/cmsis_rtos_v1/philosophers/sample.yaml
+++ b/samples/subsys/portability/cmsis_rtos_v1/philosophers/sample.yaml
@@ -1,6 +1,9 @@
 sample:
   name: CMSIS_RTOS_V1 Dining Philosophers
 common:
+  platform_key:
+    - arch
+    - simulation
   integration_platforms:
     - native_sim
   extra_args: DEBUG_PRINTF=1

--- a/samples/subsys/portability/cmsis_rtos_v1/timer_synchronization/sample.yaml
+++ b/samples/subsys/portability/cmsis_rtos_v1/timer_synchronization/sample.yaml
@@ -2,6 +2,9 @@ sample:
   name: CMSIS_RTOS_V1 Synchronization
 tests:
   sample.portability.cmsis_rtos_v1.timer_synchronization:
+    platform_key:
+      - arch
+      - simulation
     integration_platforms:
       - native_sim
     tags: cmsis_rtos

--- a/samples/subsys/portability/cmsis_rtos_v2/philosophers/sample.yaml
+++ b/samples/subsys/portability/cmsis_rtos_v2/philosophers/sample.yaml
@@ -1,6 +1,9 @@
 sample:
   name: CMSIS_RTOS_V2 Dining Philosophers
 common:
+  platform_key:
+    - arch
+    - simulation
   integration_platforms:
     - native_sim
   extra_args: DEBUG_PRINTF=1

--- a/samples/subsys/portability/cmsis_rtos_v2/timer_synchronization/sample.yaml
+++ b/samples/subsys/portability/cmsis_rtos_v2/timer_synchronization/sample.yaml
@@ -2,6 +2,9 @@ sample:
   name: CMSIS_RTOS_V2 Synchronization
 tests:
   sample.portability.cmsis_rtos_v2.timer_synchronization:
+    platform_key:
+      - arch
+      - simulation
     integration_platforms:
       - native_sim
     platform_exclude:

--- a/tests/subsys/portability/cmsis_rtos_v1/testcase.yaml
+++ b/tests/subsys/portability/cmsis_rtos_v1/testcase.yaml
@@ -1,5 +1,8 @@
 tests:
   portability.cmsis_rtos_v1:
+    platform_key:
+      - arch
+      - simulation
     tags: cmsis_rtos
     min_ram: 32
     min_flash: 34

--- a/tests/subsys/portability/cmsis_rtos_v2/testcase.yaml
+++ b/tests/subsys/portability/cmsis_rtos_v2/testcase.yaml
@@ -1,6 +1,8 @@
 tests:
   portability.cmsis_rtos_v2:
-    platform_exclude: m2gl025_miv
+    platform_key:
+      - arch
+      - simulation
     tags: cmsis_rtos
     min_ram: 32
     min_flash: 34

